src/occa/internal/lang/modes/withLauncher.cpp

```diff
@@ -168,6 +168,43 @@ namespace occa {
         return false;
       }
 
+      dim withLauncher::innerDims(functionDeclStatement &smnt) {
+        dim dims;
+        functionDeclStatement &kernelSmnt = (functionDeclStatement&) smnt.clone();
+
+        statementArray::from(kernelSmnt)
+          .flatFilterByAttribute("outer")
+          .filterByStatementType(statementType::for_)
+          .forEach([&](statement_t *forSmnt) {
+
+             forStatement *innerSmnt = getInnerMostInnerLoop((forStatement&) *forSmnt);
+             statementArray path = oklForStatement::getOklLoopPath(*innerSmnt);
+
+             dims[0] = 0;
+             dims[1] = 0;
+             dims[2] = 0;
+             int innerCount = 0;
+             const int pathCount = (int) path.length();
+             for (int i = 0; i < pathCount; ++i) {
+               forStatement &pathSmnt = *((forStatement*) path[i]);
+               oklForStatement oklForSmnt(pathSmnt);
+
+               if(pathSmnt.hasAttribute("inner")) {
+                 innerCount++;
```

Comment:
The inner iteration counts need to be inserted at the beginning of `dims` in the correct order.
```suggestion
                 innerCount++;
                 innerIndex = oklForSmnt.oklLoopIndex();
```